diff -cr orig/glib-2.22.2/config.h glib-2.22.2/config.h
*** orig/glib-2.22.2/config.h	2009-10-27 15:33:36.000000000 -0500
--- glib-2.22.2/config.h	2009-10-27 15:34:02.000000000 -0500
***************
*** 64,70 ****
  /* #undef G_ATOMIC_ALPHA */
  
  /* arm atomic implementation */
! #define G_ATOMIC_ARM 1
  
  /* cris atomic implementation */
  /* #undef G_ATOMIC_CRIS */
--- 64,76 ----
  /* #undef G_ATOMIC_ALPHA */
  
  /* arm atomic implementation */
! /* #undef G_ATOMIC_ARM */
! 
!  /* DJD: support for arm atomic spinlocks appeared in 2.14.1 but consistently 
!   * crashes DM6446 boards.  Disable until problem can be resolved.  There is
!   * currently no way to disable this with a configure option so we must patch
!   * the config.h file.
!   */
  
  /* cris atomic implementation */
  /* #undef G_ATOMIC_CRIS */
***************
*** 362,368 ****
  /* #undef HAVE_SELINUX */
  
  /* Define to 1 if you have the <selinux/selinux.h> header file. */
! /* #undef HAVE_SELINUX_SELINUX_H */
  
  /* Define to 1 if you have the `setenv' function. */
  #define HAVE_SETENV 1
--- 368,374 ----
  /* #undef HAVE_SELINUX */
  
  /* Define to 1 if you have the <selinux/selinux.h> header file. */
! #define HAVE_SELINUX_SELINUX_H 1
  
  /* Define to 1 if you have the `setenv' function. */
  #define HAVE_SETENV 1
Only in glib-2.22.2: config.h.orig
diff -cr orig/glib-2.22.2/glibconfig.h glib-2.22.2/glibconfig.h
*** orig/glib-2.22.2/glibconfig.h	2009-10-27 15:33:38.000000000 -0500
--- glib-2.22.2/glibconfig.h	2009-10-27 15:34:02.000000000 -0500
***************
*** 167,172 ****
--- 167,174 ----
    long   dummy_long;
  };
  
+ #define G_ATOMIC_OP_MEMORY_BARRIER_NEEDED 1
+ 
  #define GINT16_TO_LE(val)	((gint16) (val))
  #define GUINT16_TO_LE(val)	((guint16) (val))
  #define GINT16_TO_BE(val)	((gint16) GUINT16_SWAP_LE_BE (val))
