diff -c -r orig/gstreamer-0.10.21/docs/gst/Makefile.in gstreamer-0.10.21/docs/gst/Makefile.in
*** orig/gstreamer-0.10.21/docs/gst/Makefile.in	2008-10-02 16:54:18.000000000 -0500
--- gstreamer-0.10.21/docs/gst/Makefile.in	2008-12-08 19:04:29.000000000 -0600
***************
*** 808,815 ****
          	    $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp2 \
  	           $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
  	  fi; \
! 	  which gtkdoc-rebase >/dev/null && \
! 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
  	fi) 
  uninstall-local:
  	(installfiles=`echo ./html/*.html`; \
--- 808,815 ----
          	    $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp2 \
  	           $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
  	  fi; \
! 	  (which gtkdoc-rebase >/dev/null && \
! 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR)) || true; \
  	fi) 
  uninstall-local:
  	(installfiles=`echo ./html/*.html`; \
diff -c -r orig/gstreamer-0.10.21/docs/libs/Makefile.in gstreamer-0.10.21/docs/libs/Makefile.in
*** orig/gstreamer-0.10.21/docs/libs/Makefile.in	2008-10-02 16:54:18.000000000 -0500
--- gstreamer-0.10.21/docs/libs/Makefile.in	2008-12-08 19:04:14.000000000 -0600
***************
*** 810,817 ****
          	    $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp2 \
  	           $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
  	  fi; \
! 	  which gtkdoc-rebase >/dev/null && \
! 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
  	fi) 
  uninstall-local:
  	(installfiles=`echo ./html/*.html`; \
--- 810,817 ----
          	    $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp2 \
  	           $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
  	  fi; \
! 	  (which gtkdoc-rebase >/dev/null && \
! 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR)) || true; \
  	fi) 
  uninstall-local:
  	(installfiles=`echo ./html/*.html`; \
