diff -uNr orig/gst-plugins-base-0.10.25/gst/playback/gstplaysink.c gst-plugins-base-0.10.25/gst/playback/gstplaysink.c
--- orig/gst-plugins-base-0.10.25/gst/playback/gstplaysink.c	2009-10-01 10:25:19.000000000 -0500
+++ gst-plugins-base-0.10.25/gst/playback/gstplaysink.c	2010-01-12 08:58:30.000000000 -0600
@@ -1880,12 +1880,13 @@
      * pick one and ignore the other then instead of erroring out?) */
     if (need_text && need_subp)
       goto subs_and_text;
-  } else if (flags & GST_PLAY_FLAG_VIDEO && playsink->video_pad) {
+  } else if (((flags & GST_PLAY_FLAG_VIDEO) || 
+    (flags & GST_PLAY_FLAG_NATIVE_VIDEO)) && playsink->video_pad) {
     /* we have video and we are requested to show it */
     need_video = TRUE;
   }
   if (playsink->audio_pad) {
-    if (flags & GST_PLAY_FLAG_AUDIO) {
+    if ((flags & GST_PLAY_FLAG_AUDIO) || (flags & GST_PLAY_FLAG_NATIVE_AUDIO)) {
       need_audio = TRUE;
     }
     if (playsink->audio_pad_raw) {
